### PR TITLE
Secure the invoker with ssl.

### PR DESCRIPTION
**Pull request by cbickel**
## Description

This PR adds SSL to the invoker to secure the ping-endpoint. In addition it tidies up some controller-ssl-settings.

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [x] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [x] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [x] I updated the documentation where necessary.

### CODE REVIEW

**Comment by cbickel** [comment]
PG2#3514 🔵

File: ansible/group_vars/all [hunk 1]
```diff
@@ -83,28 +86,19 @@ controller:
...
+  protocol: "{{ controller_protocol | default('https') }}"
   ssl:
     cn: openwhisk-controllers
-    cert: "{{ controller_ca_cert | default('controller-openwhisk-server-cert.pem') }}"
-    key: "{{ controller_key | default('controller-openwhisk-server-key.pem') }}"
-    clientAuth: "{{ controller_client_auth | default('true') }}"
+    keyPrefix: "{{ __controller_ssl_keyPrefix }}"
     storeFlavor: PKCS12
+    clientAuth: "{{ controller_client_auth | default('true') }}"
+    cert: "controller-openwhisk-server-cert.pem"
```
Comment (vvraskin) [inline ansible/group_vars/all]: Should we use this var `__controller_ssl_keyPrefix` for both cert and key variables?

File: ansible/group_vars/all [hunk 2]
```diff
@@ -189,6 +186,17 @@ invoker:
     {{ jmx.jvmCommonArgs }} -Djava.rmi.server.hostname={{ invokerHostname }} -Dcom.sun.management.jmxremote.rmi.port={{ jmx.rmiBasePortInvoker + groups['invokers'].index(inventory_hostname) }} -Dcom.sun.management.jmxremote.port={{ jmx.basePortInvoker + groups['invokers'].index(inventory_hostname) }}
     {% endif %}"
   extraEnv: "{{ invoker_extraEnv | default({}) }}"
+  protocol: "{{ invoker_protocol | default('https') }}"
+  ssl:
+    cn: "openwhisk-invokers"
+    keyPrefix: "{{ __invoker_ssl_keyPrefix }}"
+    storeFlavor: "PKCS12"
+    clientAuth: "{{ invoker_client_auth | default('true') }}"
+    cert: "invoker-openwhisk-server-cert.pem"
```
Comment (vvraskin) [inline ansible/group_vars/all]: the same as above, `__invoker_ssl_keyPrefix`

File: common/scala/src/main/scala/whisk/common/Https.scala [hunk 3]
```diff
@@ -55,17 +45,17 @@ object Https {
...
     val keyManagerFactory: KeyManagerFactory = KeyManagerFactory.getInstance(keyFactoryType)
-    keyManagerFactory.init(ks, keystorePassword)
+    keyManagerFactory.init(keyStore, keystorePassword)
 
-// configure truststore
-    val truststorePassword = httpsConfig.truststorePassword.toCharArray
-    val ts: KeyStore = getCertStore(truststorePassword, httpsConfig.truststoreFlavor, httpsConfig.keystorePath)
     val trustManagerFactory: TrustManagerFactory = TrustManagerFactory.getInstance(keyFactoryType)
-    trustManagerFactory.init(ts)
+    trustManagerFactory.init(keyStore)
```
Comment (vvraskin) [inline common/scala/src/main/scala/whisk/common/Https.scala]: As discussed in person, should we add a comment saying that we use keystore as a truststore here or leave truststore configurable from the outside?

File: common/scala/src/main/scala/whisk/http/BasicHttpService.scala [hunk 4]
```diff
@@ -165,7 +165,7 @@ object BasicHttpService {
   /**
    * Starts an HTTPS route handler on given port and registers a shutdown hook.
    */
-  def startHttpsService(route: Route, port: Int, config: WhiskConfig)(implicit actorSystem: ActorSystem,
+  def startHttpsService(route: Route, port: Int, config: HttpsConfig)(implicit actorSystem: ActorSystem,
```
Comment (vvraskin) [inline common/scala/src/main/scala/whisk/http/BasicHttpService.scala]: This method is pretty similar to `startHttpService`, should we perhaps merge them to something like:

```
  def startHttpService(route: Route, port: Int, config: Option[HttpsConfig] = None)(
    implicit actorSystem: ActorSystem,
    materializer: ActorMaterializer): Unit = {

    implicit val executionContext = actorSystem.dispatcher
    val context = config.map(c => Https.connectionContext(c)).getOrElse(Http().defaultServerHttpContext)
    Http().bindAndHandle(route, "0.0.0.0", port, connectionContext = context)
  }
```

**Pull request merged**